mobile/cafe.go

```diff
@@ -18,48 +19,60 @@ func (m *Mobile) RegisterCafe(host string) error {
 }
 
 // CafeSessions calls core CafeSessions
-func (m *Mobile) CafeSessions() (string, error) {
+func (m *Mobile) CafeSessions() ([]byte, error) {
 	if !m.node.Started() {
-		return "", core.ErrStopped
+		return nil, core.ErrStopped
 	}
 
-	items, err := m.node.CafeSessions()
+	sessions, err := m.node.CafeSessions()
 	if err != nil {
-		return "", err
+		return nil, err
 	}
-	if len(items) == 0 {
-		items = make([]repo.CafeSession, 0)
+	cafeSessions := &pb.CafeSessions{
+		Values: sessions,
 	}
-	return toJSON(items)
+	bytes, err := proto.Marshal(cafeSessions)
+	if err != nil {
+		return nil, err
+	}
+	return bytes, nil
 }
 
 // CafeSession calls core CafeSession
-func (m *Mobile) CafeSession(peerId string) (string, error) {
+func (m *Mobile) CafeSession(peerId string) ([]byte, error) {
 	if !m.node.Started() {
-		return "", core.ErrStopped
+		return nil, core.ErrStopped
 	}
 
 	session, err := m.node.CafeSession(peerId)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	if session == nil {
-		return "", nil
+		return nil, nil
+	}
+	bytes, err := proto.Marshal(session)
+	if err != nil {
+		return nil, err
 	}
-	return toJSON(session)
+	return bytes, nil
 }
 
 // RefreshCafeSession calls core RefreshCafeSession
-func (m *Mobile) RefreshCafeSession(peerId string) (string, error) {
+func (m *Mobile) RefreshCafeSession(peerId string) ([]byte, error) {
 	if !m.node.Started() {
-		return "", core.ErrOffline
+		return nil, core.ErrStopped
```

Comment:
Think we were using the wrong error return type here.

Reply:
ah yeah, good eye